DiabloII.Application/Services/Hals/Suggestions/SuggestionHalService.cs

```diff
@@ -0,0 +1,74 @@
+﻿using System;
+using System.Linq;
+using System.Net.Http;
+using AutoMapper;
+using DiabloII.Application.Responses.Suggestions;
+using DiabloII.Application.Services.UserIdResolver;
+using Halcyon.HAL;
+using Microsoft.AspNetCore.Http;
+
+namespace DiabloII.Application.Services.Hals.Suggestions
+{
+    public class SuggestionHalService : BaseHalService, ISuggestionHalService
+    {
+        private readonly IMapper _mapper;
+        private readonly string _userId;
+
+        public SuggestionHalService(IUserIdResolverService userIdResolver, IHttpContextAccessor httpContextAccessor, IMapper mapper) : base (httpContextAccessor)
+        {
+            _mapper = mapper;
+            _userId = userIdResolver.Resolve();
+        }
+
+        public HALResponse AddLinks(SuggestionDto suggestion)
+        {
+            var suggestionHalResponse = ToSuggestionHalResponse(suggestion);
+            var canEditSuggestion = _userId == suggestion.CreatedBy;
+            var halResponse = ToHalResponse(suggestionHalResponse);
+
+            AddLink(halResponse, "suggestion_create", HttpMethod.Post);
+
+            if (canEditSuggestion)
+                AddLink(halResponse, "suggestion_delete", HttpMethod.Delete, suggestion.Id.ToString());
+
+            return halResponse;
+        }
+
+        // This logic mapping should be extracted, in better case should go in configuration of automapper.
+        private SuggestionHalResponse ToSuggestionHalResponse(SuggestionDto suggestion)
+        {
+            var suggestionHalResponse = _mapper.Map<SuggestionHalResponse>(suggestion);
+
+            suggestionHalResponse.Votes = suggestion.Votes.Select(vote => AddLinks(vote, suggestion.Id)).ToList();
```

Comment:
All this mapping logic should be extrated in the mapper, AutoMapper looks to don't allow dependency injection, this *** forces me to add messy code...